src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfigValueSerializer`1.cs

```diff
@@ -20,6 +20,9 @@ internal sealed class EditorConfigValueSerializer<T>(
         parseValue: _ => throw new NotSupportedException("Option does not support serialization to editorconfig format"),
         serializeValue: _ => throw new NotSupportedException("Option does not support serialization to editorconfig format"));
 
+    public readonly Func<string, Optional<T>> ParseValue = parseValue;
+    public readonly Func<T, string> SerializeValue = serializeValue;
```

Comment:
exposed so we can safely create a new instance with anotehr T type that defers to this T type.

Reply:
Why not move the convert functionality directly to this type, so we can keep this private?